plugins/wasi_nn/CMakeLists.txt

```diff
@@ -195,6 +210,14 @@ foreach(BACKEND ${WASMEDGE_PLUGIN_WASI_NN_BACKEND})
       whisper
       simdjson::simdjson
     )
+    if(APPLE AND WASMEDGE_PLUGIN_WASI_NN_WHISPER_METAL)
```

Comment:
Remove this.
For embedding metal library, the extra header and metal file are not needed.